services/QuillLMS/engines/evidence/lib/evidence/evidence/open_ai/completion.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Evidence
+  module OpenAI
+    class Completion < Evidence::ApplicationService
+      include HTTParty
+      base_uri 'https://api.openai.com/v1'
+
+      API_KEY = ENV['OPENAI_API_KEY']
+      MAX_TOKENS = 24
+      ENDPOINT = '/completions'
+      # A-D, (A)Least -> (D)Most Complex/Expensive
+      MODELS = {
+        ada: 'text-ada-001',
+        babbage: 'text-babbage-001',
+        curie: 'text-curie-001',
+        davinci: 'text-davinci-002'
+      }
+      BLANK = ''
+      STOP_TOKENS = [". ", ", "]
+
+      attr_accessor :response, :prompt, :temperature, :count, :model_key, :options_hash
+
+      def initialize(prompt:, temperature: 0.5, count: 1, model_key: :babbage, options_hash: {})
+        @prompt = prompt
+        @temperature = temperature
+        @count = count
+        @model_key = model_key
+        @options_hash = options_hash
+      end
+
+      def headers
+        {
+          "Content-Type" => "application/json",
+          "Authorization" => "Bearer #{API_KEY}"
+        }
+      end
+
+      def request_body
+        {
+          model: MODELS[model_key],
+          temperature: temperature,
+          prompt: prompt,
+          n: count,
+          max_tokens: MAX_TOKENS,
+          stop: STOP_TOKENS
+        }.merge(options_hash)
+      end
+
+      def run
+        return response if response.present?
+
+        @response = self.class.post(ENDPOINT, body: request_body.to_json, headers: headers)
+
+        cleaned_results
+      end
+
+      def cleaned_results
+        result_texts
+          .map{|r| r.gsub(/^(\n|-|\s)+/, BLANK)} # strip all leading \n, -, or whitespace
```

Comment:
It's similar, but it has a different effect, so both are needed. 

The first regex is removing starting sequences of newlines,spaces,or dashes, so if the space is removed, strings starting with `" \n"` won't be cleaned properly (especially because of the split on remaining `\n` later:

```ruby
input = "   \n\n this is a sentence \n this is a second one"
# current cleaning 
output = "this is a sentence"
# removing the \s
output = ""

```
The second strip is because some of the things removed can cause extra whitespace.